sqlmesh/core/plan/definition.py

```diff
@@ -319,6 +325,32 @@ def set_choice(self, snapshot: Snapshot, choice: SnapshotChangeCategory) -> None
         self._categorized = None
         self._uncategorized = None
 
+    @property
+    def effective_from(self) -> t.Optional[TimeLike]:
+        """The effective date for all new snapshots in the plan."""
+        return self._effective_from
+
+    @effective_from.setter
+    def effective_from(self, effective_from: t.Optional[TimeLike]) -> None:
```

Comment:
why have this and both _set_effective_from? why not just have the setter?

Reply:
So that it doesn't look confusing when we set in the constructor.

Reply:
i don't think it will be confusing?

if effective_from:
   self.effective_from = effective_from is fine right?

Reply:
It'd be more like:
```
self._effective_from = None
if effective_from:
    self.effective_from = effective_from is fine right?
```
So it looks like we're setting it twice, which is a confusing part.

Reply:
you don't need to initialize the first one

Reply:
I do, because `plan.effective_from` will return an attribute error if I never set it.